karapace/schema_models.py

```diff
@@ -116,20 +140,71 @@ def normalize_schema_str(schema_str: str, schema_type: SchemaType) -> str:
         return schema_str
 
     def __str__(self) -> str:
-        return self.schema_str
+        if self.schema_type == SchemaType.PROTOBUF:
+            return self.schema_str
+
+        if self._str_cached is None:
+            self._str_cached = json_encode(self.to_dict())
+        return self._str_cached
 
     def __repr__(self) -> str:
         return f"TypedSchema(type={self.schema_type}, schema={str(self)})"
 
+    @property
+    def schema(self) -> Union[Draft7Validator, AvroSchema, ProtobufSchema]:
+        if self._schema_cached is not None:
+            return self._schema_cached
+        if self.schema_type is SchemaType.AVRO:
```

Comment:
This method body looks equivalent to using `self._schema_cached = parse(...)`.  Instead of copying it here, we could just use that method.

Is it necessary to push the parsed schema into the base here?

Reply:
will fix